hermes/probe/delete/delete_file_test.go

```diff
@@ -0,0 +1,178 @@
+			},
+		},
+		ApiCallLatencyDistribution: &metricpb.Dist{
+			Buckets: &metricpb.Dist_ExplicitBuckets{
+				ExplicitBuckets: "0.000000002,0.000000004,0.000000008,0.000000016,0.00000032,0.000000064,0.000000128,100",
+			},
+		},
+	}
+	return hermesExtension
+}
+
+// getTargetPb generates a target proto config.
+// Returns:
+//	- target: target proto config
+func genTargetPb() *monitorpb.Target {
+	return &monitorpb.Target{
+		Name:                   "hermes",
+		TargetSystem:           monitorpb.Target_GOOGLE_CLOUD_STORAGE,
+		TotalSpaceAllocatedMib: int64(100),
+		BucketName:             bucketName,
+	}
+}
+
+const (
+	firstID    = int32(1)
+	lastID     = int32(50)
+	contents   = "abc123"
+	hash       = "6367c48dd193d56ea7b0baad25b19455e529f5ee"
+	bucketName = "test_bucket_5"
+)
+
+// genTestTarget generates an initialised test Target struct.
+func genTestTarget(cfg *monitorpb.HermesProbeDef, t *testing.T) *probe.Target {
+	filenames := make(map[int32]string)
+
+	for i := firstID; i <= lastID; i++ {
+		var id string
+		if i <= 9 {
+			id = fmt.Sprintf("0%d", i)
+		} else {
+			id = fmt.Sprintf("%d", i)
+		}
+		filenames[i] = fmt.Sprintf("Hermes_%s_%s", id, hash)
+	}
+
+	metrics, err := m.NewMetrics(cfg, genTargetPb())
+	if err != nil {
+		t.Fatalf("could not initialise metrics using config and target provided, err: %v", err)
+	}
+
+	return &probe.Target{
+		Target: cfg.GetTargets()[0],
+		Journal: &journalpb.StateJournal{
+			Intent:    &journalpb.Intent{},
+			Filenames: filenames,
+		},
+		LatencyMetrics: metrics,
+	}
+}
+
+// setupMockSystem sets up the mock storage system through the fake client.
+func setupMockSystem(ctx context.Context, t *testing.T) stiface.Client {
+	client := mock.NewFakeClient()
+
+	mockBucket := client.Bucket(bucketName)
+	if err := mockBucket.Create(ctx, "", nil); err != nil {
+		t.Fatalf("could not create mock bucket when setting up mock storage system, err: %v", err)
+	}
+
+	for i := firstID; i <= lastID; i++ {
+		var id string
+		if i <= 9 {
+			id = fmt.Sprintf("0%d", i)
+		} else {
+			id = fmt.Sprintf("%d", i)
+		}
+		filename := fmt.Sprintf("Hermes_%s_%s", id, hash)
+
+		writer := client.Bucket(bucketName).Object(filename).NewWriter(ctx)
+		n, err := writer.Write([]byte(contents))
+		if err != nil || n != len([]byte(contents)) {
+			t.Fatalf("failed to create file during setup of mock storage system, err: %v", err)
+		}
+		writer.Close()
+	}
+
+	return client
+}
+
+func TestDeleteRandomFile(t *testing.T) {
+	testProbeName := "testDelete1"
+	ctx := context.Background()
+
+	client := setupMockSystem(ctx, t)
+	target := genTestTarget(genTestConfig(testProbeName), t)
+
+	logger, err := logger.NewCloudproberLog(testProbeName)
+	if err != nil {
+		t.Fatalf("error in initializing logger for the probe (%s): %v", testProbeName, err)
+	}
+
+	fileID, err := DeleteRandomFile(ctx, target, client, logger)
+	if err != nil {
+		t.Errorf("deleteRandomFile(ID: %d) failed: expected error as %v, got %v", fileID, nil, err)
```

Comment:
Nit: Fix the name in the error message: replace `deleteRandomFile` with `DeleteRandomFile`.

Reply:
DeleteRandomFile() has been removed.